analyzers/src/SonarAnalyzer.CSharp/Rules/LockedFieldShouldBeReadonly.cs

```diff
@@ -37,27 +40,24 @@ private static void CheckLockStatement(SonarSyntaxNodeReportingContext context)
         var expression = ((LockStatementSyntax)context.Node).Expression?.RemoveParentheses();
         if (IsCreation(expression))
         {
-            ReportIssue("a new instance because is a no-op");
+            context.ReportIssue(Diagnostic.Create(LockedFieldRule, expression.GetLocation(), "a new instance because is a no-op"));
         }
         else
         {
             var lazySymbol = new Lazy<ISymbol>(() => context.SemanticModel.GetSymbolInfo(expression).Symbol);
             if (IsOfTypeString(expression, lazySymbol))
             {
-                ReportIssue("strings as they can be interned");
+                context.ReportIssue(Diagnostic.Create(LockedFieldRule, expression.GetLocation(), "strings as they can be interned"));
             }
             else if (expression is IdentifierNameSyntax && lazySymbol.Value is ILocalSymbol localSymbol)
             {
-                ReportIssue($"local variable '{localSymbol.Name}'");
+                context.ReportIssue(Diagnostic.Create(LocalVariableRule, expression.GetLocation(), $"local variable '{localSymbol.Name}'"));
             }
             else if (FieldWritable(expression, lazySymbol) is { } field)
             {
-                ReportIssue($"writable field '{field.Name}'");
+                context.ReportIssue(Diagnostic.Create(LockedFieldRule, expression.GetLocation(), $"writable field '{field.Name}'"));
             }
         }
-
-        void ReportIssue(string message) =>
```

Comment:
Nitpick: I'd keep this method and add the Rule as a parameter because it's still shorter.